packages/alfa-rules/src/sia-r66/rule.ts

```diff
@@ -35,31 +37,68 @@ export default Rule.Atomic.of<Page, Text, Question.Metadata>({
   evaluate({ device, document }) {
     return {
       applicability() {
-        return visit(document);
+        let disabledWidgetTexts: Set<Text> = Set.empty();
+        let textNodes: Sequence<Text> = Sequence.empty();
 
-        function* visit(node: Node): Iterable<Text> {
+        gather(document);
+        return getApplicableTexts();
+
+        function gather(node: Node): void {
+          // Gather all applicable text nodes
+          if (
+            node.parent().isSome() &&
```

Comment:
```suggestion
            node.parent().some(…
```
Will take a predicate as argument and automatically be `false` if the option is `None`.